cloud/shared/bin/lib/config_loader_test.py

```diff
@@ -132,6 +135,48 @@ def test_value_regex_ignored_for_not_required_and_not_provided(self):
         config_loader.configs = {}
         self.assertEqual(config_loader.validate_config(), [])
 
+    def mocked_get(url):
+        response = requests.Response()
+        response.headers["Content-type"] = "application/json"
+
+        # mock out getting te version number that matches "latest"
+        if url == "https://api.github.com/repos/civiform/civiform/releases/latest":
+            response.status_code = 200
+            data_1 = {"tag_name": "v1.23.0"}
+            response._content = json.dumps(data_1).encode()
+
+        # mock out getting the sha for for the version number v1.23.0
+        elif url == "https://api.github.com/repos/civiform/civiform/git/refs/tags/v1.23.0":
+            response.status_code = 200
+            data = {"object": {"sha": "0123456789abcdef"}}
+            response._content = json.dumps(data).encode()
+
+        # mock out the case when the version number is not available and the request fails
+        else:
+            response.status_code = 404
+            data = {"message": "no json found"}
+            response._content = json.dumps(data).encode()
+
+        return response
+
+    @patch('requests.get', side_effect=mocked_get)
+    def test_get_commit_hash_for_release__success_case(self, mocked_get):
+        config_loader = ConfigLoader()
+        commit_sha = config_loader.get_commit_sha_for_release("latest")
+
+        #mock_requests.assert_called_with("https://api.github.com/repos/civiform/civiform/git/refs/tags/V1.0.0")
+        self.assertEqual(commit_sha, "0123456789abcdef")
+
+    @patch('requests.get', side_effect=mocked_get)
+    def test_get_commit_hash_for_release__fail_case(self, mocked_requests):
+        config_loader = ConfigLoader()
+        try:
+            commit_sha = config_loader.get_commit_sha_for_release("invalid tag")
+        except ConfigLoader.VersionNotFoundError as e:
+            self.assertEqual(
+                """The commit sha for version invalid tag could not be found. Are you using a valid tag such as latest or a valid version number like v1.0.0?
+ 404 - no json found""", e.args[0])
+
```

Comment:
suggestion (non-blocking): add a test case for passing in a tag other than "latest" (could use "v1.23.0" from your mock)